src/main/java/com/amazonaws/kinesisvideo/internal/producer/StreamEventMetadata.java

```diff
@@ -39,14 +44,103 @@ public class StreamEventMetadata {
     * @param  names The custom data names, to be paired with values, length must be <= MKV_MAX_TAG_NAME_LEN as defined in the native code.
     * @param  values The custom data values, to be paired with names, length must be <= MKV_MAX_TAG_VALUE_LEN as defined in the native code.
     */
-    public StreamEventMetadata(String imagePrefix, int numberOfPairs, String[] names, String[] values) {
+    private StreamEventMetadata(String imagePrefix, int numberOfPairs, String[] names, String[] values) {
+        // Validate input parameters to prevent segmentation faults in JNI layer
+        if (numberOfPairs > 0) {
+            if (names == null) {
+                throw new IllegalArgumentException("Names array cannot be null when numberOfPairs > 0");
+            }
+            if (values == null) {
+                throw new IllegalArgumentException("Values array cannot be null when numberOfPairs > 0");
+            }
+            if (names.length < numberOfPairs) {
```

Comment:
Shouldn't these be == but both be < the max possible pairs? And same for `values.length`.